Formula/pythran.rb

```diff
@@ -0,0 +1,116 @@
+class Pythran < Formula
+  desc "Ahead of Time compiler for numeric kernels"
+  homepage "https://pythran.readthedocs.io/"
+  url "https://files.pythonhosted.org/packages/2e/e9/bc3c16d683dbba5d34666b9861d1c1fdb07da0539655a3cf23a62e4cafa1/pythran-0.9.11.tar.gz"
+  sha256 "a317f91e2aade9f6550dc3bf40b5caeb45b7e012daf27e2b3e4ad928edb01667"
+  license "BSD-3-Clause"
+
+  keg_only <<~EOS
+    this formula is mainly used internally by other formulae.
+    Users are advised to use `pip` to install pythran
+  EOS
```

Comment:
Just copied from `cython`... Need info on whether to remove or leave keg-only.

Reply:
This is probably fine, I think. Does the fact that we no longer write an env script that sets `PYTHRANRC` mean it's no longer needed?

Reply:
The `keg_only` was mainly based on `cython`, which is probably the most similar to `pythran` (i.e. Python-implemented compilers).

`PYTHRANRC` was a separate issue and shouldn't be related to this.
I removed it by switching to inreplacing systemwide config files similar to Fedora.

Reply:
It was kind of related, since an env script that sets `PYTHRANRC` might override a user's `~/.pythranrc` if they don't also set `PYTHRANRC` themselves. That would be a good reason to make it keg-only. If it's no longer needed then that's one less reason.

`cython` being keg-only originates from when it was first added: eb5b6293d24dbe005f7d5dbff324651407b4e635. @DomT4 do you remember the reasons why, and whether they apply here as well?

Reply:
Homebrew used to (or at least, more so than currently) very strongly believe in not packaging command-line tools globally that were packaged by another readily-available package manager, which is largely where the logic came from. 

More practically & less philosophically: It just reached the point where we were vendoring `cython` into so many formulae we decided to create a formula for it that would exclusively be a `:build`-only dependency and therefore mostly existed for the sake of speeding up & making consistent `cython`-using BrewTestBot builds rather than ever being _intended_ to be something users would use or even really noticed existed.

Reply:
I wonder how much of Homebrew's user base actually pays attention to these messages given `cython` has been installed ~2k/mo times on request and `sphinx-doc` has been installed ~20k/mo times on request.

Though, `cython`'s documentation specifically says to install via pip or use a distribution like Anaconda: https://cython.readthedocs.io/en/latest/src/quickstart/install.html

On the other hand, `sphinx-doc` actually mentions installing via Homebrew: https://www.sphinx-doc.org/en/master/usage/installation.html#macos

And `pythran` mentions installation on some Linux package managers for ArchLinux and Fedora: https://pythran.readthedocs.io/en/latest/#archlinux

Based on this, I guess `pythran` might be okay using the more recent approach of allowing pip/etc-installable tools to be linked unless there are conflicts/issues. It also only has 1 dependent right now, so isn't exactly in the same situation as `cython`.

Reply:
> I wonder how much of Homebrew's user base actually pays attention to these messages

Between the analytics and prior experience the answer seems to be "very little" lol. The amount of issues we and other projects used to get reported with people forcing OpenSSL to link into `/usr/local` despite being `keg_only`, until Mike made `brew` refuse to link it no matter what the user tried, makes me suspect people will essentially try to make Homebrew fit their use case regardless of our intentions for a formula/tool/etc and that'll never really change 😅.

> Based on this, I guess pythran might be okay using the more recent approach of allowing pip/etc-installable tools to be linked unless there are conflicts/issues

I think unless it's something that a lot of people are likely to have installed using `pip` and _also_ then have installed by Homebrew without their explicit knowledge/etc as a dependency, it should be okay to not be `keg_only`, IMO. At this point so many people will have installed `cython` with `pip` I don't think we'd be able to particularly smoothly remove that from `keg_only` status without running into:
```
Error: Could not symlink bin/cygdb
Target /opt/homebrew/bin/cygdb
already exists.
```
and so on, although I can't remember if `brew` added a graceful way to handle that at some point.